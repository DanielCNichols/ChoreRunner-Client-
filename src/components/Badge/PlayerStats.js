import React from 'react';
import s from './PlayerStats.module.css';
import images from '../../ImgAssets/index';

export default function Badge({
  userStats: { level_id, name, total_score, badge, toNextLevel },
}) {
  return (
    <section className={s.levelInfoGrid}>
      <h3>Runner Stats</h3>
      <div className={s.badge}>
        <img src={images[badge]} alt="Badge.png" />
        <p>Level {level_id}</p>
      </div>
      <div className={s.status}>
        <h3>{name}</h3>
        <p>
          Total EXP: <span>{total_score}</span>
        </p>
        <p>
          EXP to next level: <span>{toNextLevel}</span>
        </p>
      </div>
    </section>
  );
}
