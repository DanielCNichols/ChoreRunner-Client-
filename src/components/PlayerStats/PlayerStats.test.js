import React from 'react';
import ReactDOM from 'react-dom';
import PlayerStats from './PlayerStats';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('PlayerStats component testing', () => {
  const userStats = {
    level_id: 1,
    name: 'daniel',
    total_score: 23,
    badge: 'Badge1',
    pointsToNextLevel: 2,
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <MemoryRouter>
        <PlayerStats userStats={userStats} />
      </MemoryRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <PlayerStats userStats={userStats} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
