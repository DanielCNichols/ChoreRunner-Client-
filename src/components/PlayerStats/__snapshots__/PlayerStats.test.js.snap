// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerStats component testing renders UI as expected 1`] = `
<section
  className="levelInfoGrid"
>
  <h3
    className="videoGameTitles"
  >
    daniel
    's Stats
  </h3>
  <div
    className="badge"
  >
    <div
      className="imageContainer"
    >
      <img
        alt="Badge.png"
        src="Badge1.png"
      />
    </div>
    <p>
      Level 
      1
    </p>
  </div>
  <div
    className="status"
  >
    <h3>
      Experience
    </h3>
    <div
      className="progressBarContainer"
    >
      <div
        style={
          Object {
            "backgroundColor": "blue",
            "border": "2px solid white",
            "height": "30px",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <span
          style={
            Object {
              "color": "#ffd900",
              "display": "inline-block",
              "fontWeight": "bold",
              "padding": "5px",
              "position": "relative",
              "textAlign": "center",
              "verticalAlign": "middle",
              "width": "100%",
              "zIndex": "2",
            }
          }
        >
          8 / 10
        </span>
        <div
          style={
            Object {
              "backgroundColor": "white",
              "height": "100%",
              "position": "absolute",
              "textAlign": "right",
              "top": "0",
              "width": "80%",
            }
          }
        />
      </div>
    </div>
    <p>
      Total EXP: 
      <span>
        23
      </span>
    </p>
    <p>
      EXP to next level: 
      <span>
        2
      </span>
    </p>
  </div>
</section>
`;
